Platforms/QemuQ35Pkg/PlatformBuild.py

```diff
@@ -318,282 +327,74 @@ def __SetEsrtGuidVars(self, var_name, guid_str, desc_string):
-        if test_regex != "":
-            ut.set_test_regex(test_regex)
-            ut.find_tests()
-            ut.copy_tests_to_virtual_drive(VirtualDrive)
-
+        drive_path = self.env.GetValue("VIRTUAL_DRIVE_PATH")
+        
+        # General debugging information for users
         if run_tests:
             if test_regex == "":
-                logging.warning("No tests specified using TEST_REGEX flag but RUN_TESTS is TRUE")
-            elif not empty_drive:
-                logging.info("EMPTY_DRIVE=FALSE. This could impact your test results")
+                logging.warning("Running tests, but no Tests specified. use TEST_REGEX to specify tests to run.")
+        
+            if not empty_drive:
+                logging.info("EMPTY_DRIVE=FALSE. Old files can persist, could effect test results.")
 
             if not shutdown_after_run:
-                logging.info("SHUTDOWN_AFTER_RUN=FALSE (default). XML test results will not be \
-                    displayed until after the QEMU instance ends")
-            ut.write_tests_to_startup_nsh(startup_nsh)
+                logging.info("SHUTDOWN_AFTER_RUN=FALSE. You will need to close qemu manually to gather test results.")
+       
 
-        nshpath = os.path.join(output_base, "startup.nsh")
-        startup_nsh.WriteOut(nshpath, shutdown_after_run)
-        VirtualDrive.AddFile(nshpath)
+        # Get a reference to the virtual drive, creating / wiping as necessary
+        # Helper located at QemuPkg/Plugins/VirtualDriveManager
+        virtual_drive = self.Helper.get_virtual_drive(drive_path)
+        if empty_drive:
+            virtual_drive.wipe()
+
+        if not virtual_drive.exists():
+            virtual_drive.make_drive()
+
+        # Add tests if requested, auto run if requested
+        if test_regex != "":
+            test_list = []
+            for pattern in test_regex.split(","):
+                test_list.extend(Path(output_base, "X64").glob(pattern))
+            
+            self.Helper.add_tests(virtual_drive, test_list, auto_run = run_tests, auto_shutdown = shutdown_after_run)
 
         # Get the version number (repo release)
-        outstream = StringIO()
-        version = "Unknown"
-        ret = RunCmd('git', "rev-parse HEAD", outstream=outstream)
-        if ret == 0:
-            commithash = outstream.getvalue().strip()
+        head = repo_resolver.repo_details(self.GetWorkspaceRoot()).get("Head", None)
+        if head:
             outstream = StringIO()
-            # See git-describe docs for a breakdown of this command output
-            ret = RunCmd("git", f'describe {commithash} --tags', outstream=outstream)
+            ret = RunCmd("git", f'describe {head["HexSha"]} --tags', outstream=outstream)
             if ret == 0:
                 version = outstream.getvalue().strip()
+        else:
+            version = "unknown"
 
         self.env.SetValue("VERSION", version, "Set Version value")

```

Comment:
This was previous logic, I just changed it in this PR to use repo_details() to get the commit hash instead of RunCmd. I could move it to the qemu_runner plugin as it's the only consumer of the version.